web-api/src/models.rs

```diff
@@ -182,26 +182,31 @@ pub(crate) struct DocumentProperty(Value);
 
 impl DocumentProperty {
     pub(crate) fn try_from_value(
-        document_id: &DocumentId,
         property_id: &DocumentPropertyId,
         mut value: Value,
     ) -> Result<Self, InvalidDocumentProperty> {
+        let check_string_property_value = |s: &str| {
+            if s.len() <= 2_048 && GENERIC_STRING_SYNTAX.is_match(s) {
+                Ok(())
+            } else {
+                Err(())
+            }
+        };
```

Comment:
🔧 we can reuse the `validate_string()` function
```suggestion
        let validate_string =
            |value: &str| validate_string(value, 0..=2_048, &GENERIC_STRING_SYNTAX);
```